api/app/models/member.rb

```diff
@@ -9,16 +9,21 @@ class Member < ApplicationRecord
   validates :role, presence: true, inclusion: { in: %w[admin participant] }
 
   # Before
-  before_destroy :destroy_associated_sessions
+  before_destroy :assign_new_session_creator
 
   def promote!
     update!(role: 'admin')
   end
 
   private
 
-  def destroy_associated_sessions
+  def assign_new_session_creator
     sessions_created_by_member = Session.where(creator_id: id)
-    sessions_created_by_member.destroy_all
+    admin_member = group.members.where(role: 'admin').where.not(id:).first
```

Comment:
Tenemos un método `admins` en el modelo de `Group` que trae todos los admins del grupo.

Te quedaría más simplificado esto 🙂 
```suggestion
    admin_member = group.admins.where.not(id:).first
```